tree/ntuple/v7/inc/ROOT/RPageStorage.hxx

```diff
@@ -276,6 +277,53 @@ mapped into memory. The page source also gives access to the ntuple's meta-data.
 */
 // clang-format on
 class RPageSource : public RPageStorage {
+public:
+   /// An RAII wrapper used for the read-only access to RPageSource::fDescriptor
+   class RSharedDescriptorGuard {
+      const RNTupleDescriptor &fDescriptor;
+      std::shared_mutex &fLock;
+
+   public:
+      RSharedDescriptorGuard(const RNTupleDescriptor &desc, std::shared_mutex &lock) : fDescriptor(desc), fLock(lock)
+      {
+         fLock.lock_shared();
+      }
+      RSharedDescriptorGuard(const RSharedDescriptorGuard &) = delete;
+      RSharedDescriptorGuard &operator=(const RSharedDescriptorGuard &) = delete;
+      RSharedDescriptorGuard(RSharedDescriptorGuard &&) = delete;
```

Comment:
Since the RSharedDescriptorGuard (and RExclDescriptorGuard) are returned by value, wouldn't it be "much more" efficient to implement the move constructor?   

Humm .. right .. since the other constructors (including implicitly the no argument constructor) are deleted, the only to actually return those objects by value is to rely on the in-place constructor of the returnee.